diff -ruN openvpn-2.6.12/distro/systemd/openvpn-client@.service.in openvpn-patch/distro/systemd/openvpn-client@.service.in
--- openvpn-2.6.12/distro/systemd/openvpn-client@.service.in	2024-07-17 23:14:46.000000000 +0300
+++ openvpn-patch/distro/systemd/openvpn-client@.service.in	2026-05-08 09:30:40.831923914 +0300
@@ -11,6 +11,9 @@
 PrivateTmp=true
 WorkingDirectory=/etc/openvpn/client
 ExecStart=@sbindir@/openvpn --suppress-timestamps --nobind --config %i.conf
+User=openvpn
+Group=network
+AmbientCapabilities=CAP_IPC_LOCK CAP_NET_ADMIN CAP_NET_RAW CAP_SETGID CAP_SETUID CAP_SETPCAP CAP_SYS_CHROOT CAP_DAC_OVERRIDE
 CapabilityBoundingSet=CAP_IPC_LOCK CAP_NET_ADMIN CAP_NET_RAW CAP_SETGID CAP_SETUID CAP_SETPCAP CAP_SYS_CHROOT CAP_DAC_OVERRIDE
 LimitNPROC=10
 DeviceAllow=/dev/null rw
diff -ruN openvpn-2.6.12/distro/systemd/openvpn-server@.service.in openvpn-patch/distro/systemd/openvpn-server@.service.in
--- openvpn-2.6.12/distro/systemd/openvpn-server@.service.in	2024-07-17 23:14:46.000000000 +0300
+++ openvpn-patch/distro/systemd/openvpn-server@.service.in	2026-05-08 09:30:40.833307977 +0300
@@ -11,6 +11,9 @@
 PrivateTmp=true
 WorkingDirectory=/etc/openvpn/server
 ExecStart=@sbindir@/openvpn --status %t/openvpn-server/status-%i.log --status-version 2 --suppress-timestamps --config %i.conf
+User=openvpn
+Group=network
+AmbientCapabilities=CAP_IPC_LOCK CAP_NET_ADMIN CAP_NET_BIND_SERVICE CAP_NET_RAW CAP_SETGID CAP_SETUID CAP_SETPCAP CAP_SYS_CHROOT CAP_DAC_OVERRIDE CAP_AUDIT_WRITE
 CapabilityBoundingSet=CAP_IPC_LOCK CAP_NET_ADMIN CAP_NET_BIND_SERVICE CAP_NET_RAW CAP_SETGID CAP_SETUID CAP_SETPCAP CAP_SYS_CHROOT CAP_DAC_OVERRIDE CAP_AUDIT_WRITE
 LimitNPROC=10
 DeviceAllow=/dev/null rw
diff -ruN openvpn-2.6.12/src/openvpn/console_systemd.c openvpn-patch/src/openvpn/console_systemd.c
--- openvpn-2.6.12/src/openvpn/console_systemd.c	2024-07-17 23:14:46.000000000 +0300
+++ openvpn-patch/src/openvpn/console_systemd.c	2026-05-08 09:30:40.830405468 +0300
@@ -71,6 +71,8 @@
     }
 #endif
     argv_printf_cat(&argv, "--icon network-vpn");
+    argv_printf_cat(&argv, "--timeout=0");
+    argv_printf_cat(&argv, "--system");
     argv_printf_cat(&argv, "%s", prompt);
 
     if ((std_out = openvpn_popen(&argv, NULL)) < 0)
