diff -ruN openvpn-2.7.4/src/openvpn/manage.h ov274-aws/src/openvpn/manage.h
--- openvpn-2.7.4/src/openvpn/manage.h	2026-05-08 10:05:25.420750821 +0300
+++ ov274-aws/src/openvpn/manage.h	2026-05-08 12:08:17.699495822 +0300
@@ -59,6 +59,9 @@
 #define MANAGEMENT_SOCKET_READ_BUFFER_SIZE OPTION_PARM_SIZE
 #define MANAGEMENT_STATE_BUFFER_SIZE        100
 
+#define COMMAND_LINE_OPTION_BUFFER_SIZE OPTION_PARM_SIZE
+#define MANAGEMENT_SOCKET_READ_BUFFER_SIZE OPTION_PARM_SIZE
+
 /*
  * Management-interface-based deferred authentication
  */
diff -ruN openvpn-2.7.4/src/openvpn/ssl.c ov274-aws/src/openvpn/ssl.c
--- openvpn-2.7.4/src/openvpn/ssl.c	2026-04-30 16:45:57.000000000 +0300
+++ ov274-aws/src/openvpn/ssl.c	2026-05-08 12:11:29.551502641 +0300
@@ -1781,7 +1781,7 @@
 static bool
 write_empty_string(struct buffer *buf)
 {
-    if (!buf_write_u16(buf, 0))
+    if (!buf_write_u32(buf, 0))
     {
         return false;
     }
@@ -1792,12 +1792,12 @@
 write_string(struct buffer *buf, const char *str, const int maxlen)
 {
     const size_t len = strlen(str) + 1;
-    const size_t real_maxlen = (maxlen >= 0 && maxlen <= UINT16_MAX) ? (size_t)maxlen : UINT16_MAX;
+    const size_t real_maxlen = (maxlen >= 0) ? (size_t)maxlen : SIZE_MAX;
     if (len > real_maxlen)
     {
         return false;
     }
-    if (!buf_write_u16(buf, (uint16_t)len))
+    if (!buf_write_u32(buf, (uint32_t)len))
     {
         return false;
     }
@@ -2167,6 +2167,10 @@
          * are a P2P client running in tls-server mode */
         p2p_mode_ncp(multi, session);
     }
+    // Write key length in the first 4 octets of the buffer.
+    uint32_t length = BLEN(buf);
+    memcpy(buf->data, &length, sizeof(length));
+
 
     return true;
 
