diff -ruN openvpn-2.7.4/src/openvpn/manage.h ov274-aws/src/openvpn/manage.h
--- openvpn-2.7.4/src/openvpn/manage.h	2026-05-08 10:05:25.420750821 +0300
+++ ov274-aws/src/openvpn/manage.h	2026-05-08 12:08:17.699495822 +0300
@@ -59,6 +59,9 @@
 #define MANAGEMENT_SOCKET_READ_BUFFER_SIZE OPTION_PARM_SIZE
 #define MANAGEMENT_STATE_BUFFER_SIZE        100
 
+#define COMMAND_LINE_OPTION_BUFFER_SIZE OPTION_PARM_SIZE
+#define MANAGEMENT_SOCKET_READ_BUFFER_SIZE OPTION_PARM_SIZE
+
 /*
  * Management-interface-based deferred authentication
  */
